JSQMessagesViewController/Controllers/JSQMessagesViewController.m

```diff
@@ -681,14 +681,15 @@ - (CGSize)collectionView:(UICollectionView *)collectionView
 
 - (BOOL)collectionView:(JSQMessagesCollectionView *)collectionView shouldShowMenuForItemAtIndexPath:(NSIndexPath *)indexPath
 {
+    // Positioning the edit-menu properly requires tracking which item is selected.
+    self.selectedIndexPathForMenu = indexPath;
+
     //  disable menu for media messages
     id<JSQMessageData> messageItem = [collectionView.dataSource collectionView:collectionView messageDataForItemAtIndexPath:indexPath];
     if ([messageItem isMediaMessage]) {
         return NO;
     }
 
-    self.selectedIndexPathForMenu = indexPath;
```

Comment:
I'm surprised this works since it moves it above a `return NO` which I would assume would prevent the edit menu from appearing.